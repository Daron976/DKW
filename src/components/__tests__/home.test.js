import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { render, screen, waitFor } from '@testing-library/react';
import Home from '../home';

describe('Homepage', () => {
  test('Homepage should render as expected', async () => {
    const snapHomepage = renderer
      .create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      )
      .toJSON();

    expect(snapHomepage).toMatchSnapshot();
  });

  test('Homepage loads and displays relevant text', async () => {
    render(<Home />);

    const displayedText = screen.getByText('Daron Wiafe');

    waitFor(() => {
      expect(displayedText).toBeInTheDocument();
    });
  });
});
