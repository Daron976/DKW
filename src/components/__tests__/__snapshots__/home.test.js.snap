// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Homepage should render as expected 1`] = `
<section
  className="home-content flex"
  data-testid="home"
>
  <div
    className="home-header"
  >
    <h1
      className="name-animation"
    >
      Daron Wiafe
    </h1>
    <p
      className="subtext-animation subtext"
    >
      Software developer
    </p>
    <p
      className="header-msg"
    >
      Developing more than just your technical products
    </p>
  </div>
</section>
`;
